### ISSUE TITLE
Dependency wrongly moved task with no reason

### ISSUE THREAD

**Issue by chuckn0rris** [bryntum/support#8745]
[Forum post](https://forum.bryntum.com/viewtopic.php?f=43&t=28242&p=141935#p141935)

Hello,

I have a question regarding the end date of a task. In the image ![Screenshot 2024-03-05 102458.png](https://forum.bryntum.com/download/file.php?id=27229), it is shown that "Task 33" and "Task 34" are at the end of the project. However, they should start "as early as possible". We have another Gantt model where the plan actually looks different, and both tasks do not start at the end of the project but after Task 32. Attached, you will find the complete planning as a JSON file. Please let me know how this difference can occur and why "as early as possible" is at the end of the project.
![Screenshot 2024-03-05 102458.png](https://forum.bryntum.com/download/file.php?id=27229)
[planning.json](https://github.com/bryntum/support/files/14498794/planning.json)

**Comment by arcady-zherdev** [comment]
A shorter dataset for the case is:
```json
{
  "success": true,
  "project": {
    "startDate": "2024-03-06"
  },
  "tasks": {
    "rows": [
      {
        "id": "1",
        "startDate": "2024-03-06",
        "duration": 1,
        "name": "Task 1",
        "direction": "Forward"
      },
      {
        "id": "2",
        "startDate": "2024-03-07",
        "duration": 1,
        "name": "Task 2",
        "direction": "Backward"
      },
      {
        "id": "3",
        "startDate": "2024-03-08",
        "duration": 1,
        "name": "Task 3",
        "direction": "Forward"
      },
      {
        "id": "4",
        "startDate": "2024-03-08",
        "duration": 2,
        "name": "Task 4",
        "direction": "Forward"
      }
    ]
  },
  "dependencies": {
    "rows": [
      {
        "from": "1",
        "to": "2"
      },
      {
        "from": "2",
        "to": "3"
      },
      {
        "from": "2",
        "to": "4"
      }
    ]
  }
}
```
`Task 3` is adjusted to the project end and starts Mar 9th (happens since it has an ALAP predecessor). But it can start a day earlier actually since it's not critical.

**Comment by canonic-epicure** [comment]
As discussed with Arcady, the fix for this issue is probably for 6.0.0 release, as it changes some internal engine APIs (which could be used by the users)

**Comment by canonic-epicure** [comment]
Re-targeted to `master` (6.0.0 release)
